src/HttpPendingRequest.php

```diff
@@ -14,10 +14,12 @@ class HttpPendingRequest extends PendingRequest
 {
     /**
      * Constructor
+     *
+     * @param array<int, callable> $middleware
      */
-    public function __construct(Factory $factory = null)
+    public function __construct(Factory $factory = null, $middleware = [])
```

Comment:
Are we able to add the `array` type hint here, or is PHP complaining because it extends the `PendingRequest` class?

Reply:
Mr. ChatGPT says:

> Yes, in PHP 8.1 and later, you can narrow down the type of a parameter in a subclass's constructor (or any method) as long as the new type is compatible with the parent class's type. This is part of PHP's support for contravariance in parameter types and covariance in return types.

though PHP docs just talk about contravariance which is actually the other way round if we can believe that sentence:

https://www.php.net/manual/en/language.oop5.variance.php
> Whereas, contravariance allows a parameter type to be less specific in a child method, than that of its parent.

So what about covariance on param types? Is it just me that has a hard time understanding? If it's not in the docs, then we just trust PHPStorm/Phpstan's linting, and we all want types! Please add the `array` type.

Reply:
You can keep both! So if you put the array type in the method, you can then define a further array shape in the phpdoc comment! 🔥

Reply:
Yes, like it. please add the `array` prop type hint & let's merge it!

Reply:
Done @onlime